crates/libcgroups/src/v2/devices/bpf.rs

```diff
@@ -1,117 +1,124 @@
-use anyhow::{bail, Result};
-use std::os::unix::io::RawFd;
-use std::ptr;
-
-// FIXME: add tests
-
-pub fn prog_load(license: &str, insns: &[u8]) -> Result<RawFd> {
-    let insns_cnt = insns.len() / std::mem::size_of::<libbpf_sys::bpf_insn>();
-    let insns = insns as *const _ as *const libbpf_sys::bpf_insn;
-
-    let prog_fd = unsafe {
-        libbpf_sys::bpf_load_program(
-            libbpf_sys::BPF_PROG_TYPE_CGROUP_DEVICE,
-            insns,
-            insns_cnt as u64,
-            license as *const _ as *const i8,
-            0,
-            ptr::null_mut::<i8>(),
-            0,
-        )
-    };
-
-    if prog_fd < 0 {
-        return Err(errno::errno().into());
-    }
-    Ok(prog_fd)
-}
-
+#[derive(Clone)]
 pub struct ProgramInfo {
     pub id: u32,
     pub fd: i32,
 }
 
-pub fn prog_query(cgroup_fd: RawFd) -> Result<Vec<ProgramInfo>> {
-    let mut prog_ids: Vec<u32> = vec![0_u32; 64];
-    let mut attach_flags = 0_u32;
-    for _ in 0..10 {
-        let mut prog_cnt = prog_ids.len() as u32;
-        let ret = unsafe {
-            libbpf_sys::bpf_prog_query(
-                cgroup_fd,
-                libbpf_sys::BPF_CGROUP_DEVICE,
+#[cfg_attr(test, automock)]
+pub mod prog {
+    use super::ProgramInfo;
+    use anyhow::{bail, Result};
+    use std::os::unix::io::RawFd;
+    use std::ptr;
+
+    pub fn load(license: &str, insns: &[u8]) -> Result<RawFd> {
```

Comment:
changed names to remove stuttering eg `bpf::prog::prog_load` -> `bpf::prog::load`